src/psyclone/tests/psyir/transformations/nan_test.py

```diff
@@ -0,0 +1,140 @@
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: J. Henrichs, Bureau of Meteorology
+# -----------------------------------------------------------------------------
+
+''' Module containing tests for nanTestTrans and NanTestNode
+'''
+
+from __future__ import absolute_import
+
+import pytest
+
+from psyclone.errors import InternalError
+from psyclone.psyir.nodes import (colored, Node, NanTestNode,
+                                  SCHEDULE_COLOUR_MAP)
+from psyclone.psyir.transformations import (NanTestTrans,
+                                            TransformationError)
+from psyclone.tests.utilities import get_invoke
+from psyclone.transformations import OMPParallelLoopTrans
+
+# --------------------------------------------------------------------------- #
+# ================== Extract Transformation tests =========================== #
+# --------------------------------------------------------------------------- #
+
+
+def test_extract_trans():
+    '''Tests basic functions in NanTestTrans.'''
+    nan_test = NanTestTrans()
+    assert str(nan_test) == "Create a sub-tree of the PSyIR that has " \
+                            "a node of type NanTestNode at its root."
+    assert nan_test.name == "NanTestTrans"
+
+
+# -----------------------------------------------------------------------------
+def test_malformed_extract_node(monkeypatch):
+    ''' Check that we raise the expected error if a NanTestNode does
+    not have a single Schedule node as its child. '''
+    read_node = NanTestNode()
+    monkeypatch.setattr(read_node, "_children", [])
+    with pytest.raises(InternalError) as err:
+        _ = read_node.nan_test_body
+    assert "malformed or incomplete. It should have a " in str(err.value)
+    monkeypatch.setattr(read_node, "_children", [Node(), Node()])
+    with pytest.raises(InternalError) as err:
+        _ = read_node.nan_test_body
+    assert "malformed or incomplete. It should have a " in str(err.value)
+
+
+# -----------------------------------------------------------------------------
+def test_nan_test_basic(capsys):
+    '''Check basic functionality: node names, schedule view.
+    '''
+    _, invoke = get_invoke("test11_different_iterates_over_one_invoke.f90",
+                           "gocean1.0", idx=0, dist_mem=False)
+    nan_test = NanTestTrans()
+    new_sched, _ = nan_test.apply(invoke.schedule[0].loop_body[0])
+    new_sched.view()
+    result, _ = capsys.readouterr()
+
+    # Create the coloured text (if required)
+    read_node = colored("NanTest",
+                        SCHEDULE_COLOUR_MAP["NanTest"])
+    sched_node = colored("Schedule", SCHEDULE_COLOUR_MAP["Schedule"])
+    assert """{0}[]
+            0: {1}[]
+                {0}[]""".format(sched_node, read_node) in result
+
+    read_node = new_sched[0].loop_body[0]
+    assert read_node.dag_name == "nan_test_0"
```

Comment:
Is it possible to test that the generated code compiles? If that's significant work, please create a new Issue.

Reply:
Not easy, since it would need the psydata wrapper library to be compiled. I'll consider this when doing #991.